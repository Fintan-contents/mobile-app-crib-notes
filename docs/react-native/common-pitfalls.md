---
title: ハマりがちな落とし穴
---

React Nativeで開発する際に起きた困り事や不具合についてまとめています。
