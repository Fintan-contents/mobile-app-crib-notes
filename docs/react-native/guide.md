---
title: 何らかのガイドを書きたい気持ちはある
---
