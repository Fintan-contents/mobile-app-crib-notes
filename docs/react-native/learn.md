---
title: About
hide_table_of_contents: true
---

このコンテンツはReact Nativeでのモバイルアプリケーション開発を学習するためのコンテンツです。

ReactでWebアプリケーションを実装できるがReact Nativeでモバイルアプリケーションを実装したことがないアプリケーション開発者を対象として、次のことについて学ぶことを目的としています。

- ReactとReact Nativeの違い
- React Nativeで作成したモバイルアプリケーションの動作確認方法

また、作成したモバイルアプリケーションの配布方法やテスト手法についてはこのコンテンツでは触れません。

<!-- textlint-disable ja-technical-writing/sentence-length,ja-technical-writing/max-comma,ja-spacing/ja-no-space-around-parentheses,jtf-style/3.3.かっこ類と隣接する文字の間のスペースの有無,ja-technical-writing/ja-no-mixed-period,ja-technical-writing/no-unmatched-pair -->

import {PageList} from '../../src/components';
const overviews = [
  {
    title: 'Getting Started',
    to: '/react-native/learn/getting-started',
    summary: <>React Nativeでのアプリ開発に必要な環境を用意します。</>
  },
  {
    title: 'アプリの実装前に',
    to: '/react-native/learn/basic-concepts',
    summary: <>React Nativeでアプリを実装し始める前に最低限必要となる前提知識や、React NativeとReact Navigationの基本的な知識について学びます。</>
  },
  {
    title: 'ToDoアプリの実装',
    to: '/react-native/learn/todo-app',
    summary: <>React Nativeで簡単なToDoアプリを実装します。</>
  },
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length,ja-technical-writing/max-comma,ja-spacing/ja-no-space-around-parentheses,jtf-style/3.3.かっこ類と隣接する文字の間のスペースの有無,ja-technical-writing/ja-no-mixed-period,ja-technical-writing/no-unmatched-pair -->
