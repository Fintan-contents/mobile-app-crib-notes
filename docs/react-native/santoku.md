---
title: Example Application
sidebar_label: About
hide_table_of_contents: true
---

React Nativeサンプルアプリケーションの設計ドキュメントです。

<!-- textlint-disable ja-technical-writing/sentence-length -->

import {PageList} from '../../src/components';
const overviews = [
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length -->
