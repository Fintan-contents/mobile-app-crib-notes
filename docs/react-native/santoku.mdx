---
title: Example Application
sidebar_label: About
hide_table_of_contents: true
---

<!-- textlint-disable ja-technical-writing/sentence-length -->

import {PageList} from '../../src/components';
const overviews = [
  {
    title: 'Development',
    summary: '開発環境やビルドに関するドキュメント',
    to: '/react-native/santoku/development',
  }
]

<PageList overviews={overviews} colSize={12} />

<!-- textlint-enable ja-technical-writing/sentence-length -->
