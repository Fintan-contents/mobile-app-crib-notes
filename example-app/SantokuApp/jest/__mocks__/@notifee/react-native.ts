import notifee from '@notifee/react-native';
export * from '@notifee/react-native/dist/types/Library';
export * from '@notifee/react-native/dist/types/Notification';
export * from '@notifee/react-native/dist/types/Trigger';
export * from '@notifee/react-native/dist/types/NotificationIOS';
export * from '@notifee/react-native/dist/types/NotificationAndroid';

const mock: typeof notifee = {
  cancelAllNotifications: jest.fn(),
  cancelDisplayedNotifications: jest.fn(),
  cancelTriggerNotifications: jest.fn(),
  cancelNotification: jest.fn(),
  cancelDisplayedNotification: jest.fn(),
  cancelTriggerNotification: jest.fn(),
  createChannel: jest.fn(),
  createChannels: jest.fn(),
  createChannelGroup: jest.fn(),
  createChannelGroups: jest.fn(),
  deleteChannel: jest.fn(),
  deleteChannelGroup: jest.fn(),
  displayNotification: jest.fn(),
  openAlarmPermissionSettings: jest.fn(),
  createTriggerNotification: jest.fn(),
  getTriggerNotificationIds: jest.fn(),
  getDisplayedNotifications: jest.fn(),
  getTriggerNotifications: jest.fn(),
  getChannel: jest.fn(),
  isChannelCreated: jest.fn(),
  isChannelBlocked: jest.fn(),
  getChannels: jest.fn(),
  getChannelGroup: jest.fn(),
  getChannelGroups: jest.fn(),
  getInitialNotification: jest.fn(),
  onBackgroundEvent: jest.fn(),
  onForegroundEvent: jest.fn(),
  openNotificationSettings: jest.fn(),
  registerForegroundService: jest.fn(),
  stopForegroundService: jest.fn(),
  requestPermission: jest.fn(),
  setNotificationCategories: jest.fn(),
  getNotificationCategories: jest.fn(),
  getNotificationSettings: jest.fn(),
  getBadgeCount: jest.fn(),
  setBadgeCount: jest.fn(),
  incrementBadgeCount: jest.fn(),
  decrementBadgeCount: jest.fn(),
  openBatteryOptimizationSettings: jest.fn(),
  isBatteryOptimizationEnabled: jest.fn(),
  getPowerManagerInfo: jest.fn(),
  openPowerManagerSettings: jest.fn(),
  hideNotificationDrawer: jest.fn(),
  SDK_VERSION: '5.4.1',
};

// 複数のファイルでnotifeeをimportしていた場合に、redefineされないようにモックが存在するか確認
if (!__mocks.notifee) {
  Object.defineProperty(__mocks, 'notifee', {value: mock});
}

// テストケースごとにモックは初期化しておく。
beforeEach(() =>
  Object.values(mock).forEach(fn => {
    if (jest.isMockFunction(fn)) {
      fn.mockClear();
    }
  }),
);

export default mock;
