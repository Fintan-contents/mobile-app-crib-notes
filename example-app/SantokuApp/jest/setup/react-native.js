// https://reactnavigation.org/docs/testing#mocking-native-modules
// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

// !!! NOTICE !!!
// The React Navigation documentation says that `react-native-reanimated` and `react-native-gesture-handler`
// also need to be configured with mocks.
// But since React Navigation v6 does not already depend on those packages, mocking those packages is not necessary
// for React Navigation v6.
//
// If you want to use `react-native-reanimated` and `react-native-gesture-handler`, add mock according to the guide
// provided by each package.
// - `react-native-reanimated`: https://docs.swmansion.com/react-native-reanimated/docs/guide/testing/
// - `react-native-gesture-handler`: https://docs.swmansion.com/react-native-gesture-handler/docs/guides/testing
