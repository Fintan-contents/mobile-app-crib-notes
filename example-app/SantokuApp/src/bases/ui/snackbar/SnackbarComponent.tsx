import React, {useCallback, useRef, useState} from 'react';
import {Animated, StyleProp, StyleSheet, Text, TextStyle, TouchableOpacity, View, ViewStyle} from 'react-native';

import CompositeAnimation = Animated.CompositeAnimation;

export type SnackbarShowProps = {
  /**
   * Displayed message.
   */
  message: string;
  /**
   * Style of message.
   */
  messageTextStyle?: StyleProp<TextStyle>;
  /**
   * Style to adjust the look of the snackbar.
   */
  style?: StyleProp<ViewStyle>;
  /**
   * Adjust the position of the snackbar style.
   */
  positionStyle?: StyleProp<ViewStyle>;
  /**
   * Button text.
   * Placed to the right of the message.
   */
  actionText?: string;
  /**
   * Function called when the user taps the button.
   */
  actionHandler?: () => void;
  /**
   * Style of Button text.
   */
  actionTextStyle?: StyleProp<TextStyle>;
  /**
   * Time from when the snackbar appears to when it starts to fade-out.
   */
  autoHideDuration?: number;
  /**
   * Snackbar fade-in time.
   */
  fadeInDuration?: number;
  /**
   * Snackbar fade-out time.
   */
  fadeOutDuration?: number;
  /**
   * Fade-out time to close the previously displayed snack bar when displaying a new one.
   * This is applied when you try to display another snackbar while a snackbar is being displayed.
   */
  forceFadeOutDuration?: number;
  /**
   * Timestamp
   */
  timestamp?: number;
};

export type SnackbarHideProps = {
  /**
   * Specify when to hide the snackbar.
   */
  hide?: true;
  /**
   * Fade-out time when hide is specified.
   */
  hideFadeOutDuration?: number;
};

export type SnackbarProps = SnackbarShowProps & SnackbarHideProps;

export const SnackbarComponent: React.FC<SnackbarProps> = props => {
  const fadeAnim = useRef(new Animated.Value(0)).current;
  const fadeInAnimationRef = useRef<CompositeAnimation>();
  const fadeOutAnimationRef = useRef<CompositeAnimation>();
  const barrierFadeOutAnimationRef = useRef<CompositeAnimation>();
  const [visibleSnackbarProps, setVisibleSnackbarProps] = useState<SnackbarShowProps>();

  const animationStart = useCallback(
    (
      animationRef: React.MutableRefObject<CompositeAnimation | undefined>,
      config: Animated.TimingAnimationConfig,
      callback?: Animated.EndCallback,
    ) => {
      const animation = Animated.timing(fadeAnim, config);
      animationRef.current = animation;
      animation.start(result => {
        animationRef.current = undefined;
        if (callback) {
          callback(result);
        }
      });
    },
    [fadeAnim],
  );

  const show = useCallback(() => {
    setVisibleSnackbarProps({
      actionHandler: props.actionHandler,
      actionText: props.actionText,
      actionTextStyle: props.actionTextStyle,
      autoHideDuration: props.autoHideDuration,
      fadeInDuration: props.fadeInDuration,
      fadeOutDuration: props.fadeOutDuration,
      forceFadeOutDuration: props.forceFadeOutDuration,
      message: props.message,
      messageTextStyle: props.messageTextStyle,
      positionStyle: props.positionStyle,
      style: props.style,
      timestamp: props.timestamp,
    });

    const fadeInAnimationConfig = {
      toValue: 1,
      duration: props.fadeInDuration,
      useNativeDriver: true,
    };
    animationStart(fadeInAnimationRef, fadeInAnimationConfig, ({finished}) => {
      if (finished) {
        const fadeOutAnimationConfig = {
          toValue: 0,
          delay: props.autoHideDuration,
          duration: props.fadeOutDuration,
          useNativeDriver: true,
        };
        animationStart(fadeOutAnimationRef, fadeOutAnimationConfig, () => {
          if (!barrierFadeOutAnimationRef.current) {
            setVisibleSnackbarProps(undefined);
          }
        });
      } else {
        if (!barrierFadeOutAnimationRef.current) {
          setVisibleSnackbarProps(undefined);
        }
      }
    });
  }, [
    animationStart,
    props.actionHandler,
    props.actionText,
    props.actionTextStyle,
    props.autoHideDuration,
    props.fadeInDuration,
    props.fadeOutDuration,
    props.forceFadeOutDuration,
    props.message,
    props.messageTextStyle,
    props.positionStyle,
    props.style,
    props.timestamp,
  ]);

  const forceFadeout = useCallback(
    (fadeOutDuration?: number, callback?: () => void) => {
      const barrierFadeOutAnimationConfig = {
        toValue: 0,
        duration: fadeOutDuration,
        useNativeDriver: true,
      };

      barrierFadeOutAnimationRef.current?.stop();

      animationStart(barrierFadeOutAnimationRef, barrierFadeOutAnimationConfig, ({finished}) => {
        if (finished) {
          setVisibleSnackbarProps(undefined);
          callback?.();
        }
      });
      fadeInAnimationRef.current?.stop();
      fadeOutAnimationRef.current?.stop();
    },
    [animationStart],
  );

  React.useEffect(() => {
    if (props.hide) {
      forceFadeout(props.hideFadeOutDuration);
      return;
    }
    if (!props.message) {
      return;
    }
    if (barrierFadeOutAnimationRef.current) {
      forceFadeout(props.forceFadeOutDuration, show);
      return;
    }
    if (fadeInAnimationRef.current || fadeOutAnimationRef.current) {
      forceFadeout(props.forceFadeOutDuration, show);
      return;
    }
    show();
  }, [show, forceFadeout, props.hide, props.message, props.hideFadeOutDuration, props.forceFadeOutDuration]);

  const snackbarStyle = StyleSheet.flatten([styles.snackbar, props.style]);

  const animatedViewStyle = StyleSheet.flatten<ViewStyle>([styles.animatedContainer, props.positionStyle]);
  if (animatedViewStyle?.top === undefined && animatedViewStyle.bottom === undefined) {
    animatedViewStyle.bottom = 20;
  }

  const messageTextStyle = StyleSheet.flatten([styles.messageText, props.messageTextStyle]);
  const actionTextStyle = StyleSheet.flatten([styles.actionText, props.actionTextStyle]);

  if (!visibleSnackbarProps) {
    return null;
  }

  return (
    <Animated.View style={StyleSheet.flatten([{opacity: fadeAnim}, animatedViewStyle])} testID="snackbarAnimatedView">
      <View style={snackbarStyle} testID="snackbarStyleView">
        <View style={styles.messageContainer}>
          <Text style={messageTextStyle}>{visibleSnackbarProps.message}</Text>
        </View>
        {visibleSnackbarProps.actionText && visibleSnackbarProps.actionHandler && (
          <View style={styles.actionContainer}>
            <TouchableOpacity onPress={visibleSnackbarProps.actionHandler}>
              <Text style={actionTextStyle}>{visibleSnackbarProps.actionText}</Text>
            </TouchableOpacity>
          </View>
        )}
      </View>
    </Animated.View>
  );
};

SnackbarComponent.defaultProps = {
  autoHideDuration: 4000,
  fadeInDuration: 1000,
  fadeOutDuration: 1000,
  forceFadeOutDuration: 300,
  hideFadeOutDuration: 300,
};

const styles = StyleSheet.create({
  animatedContainer: {
    flex: 1,
    position: 'absolute',
    minHeight: 48,
    paddingHorizontal: 10,
    width: '100%',
  },
  snackbar: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
    backgroundColor: '#393939',
  },
  messageContainer: {
    flex: 1,
    paddingVertical: 20,
    paddingLeft: 20,
    paddingRight: 10,
  },
  actionContainer: {
    flex: 0,
    justifyContent: 'flex-end',
    alignItems: 'center',
    paddingVertical: 20,
    paddingLeft: 10,
    paddingRight: 20,
  },
  messageText: {
    color: 'white',
  },
  actionText: {
    color: '#85A3F1',
  },
});
