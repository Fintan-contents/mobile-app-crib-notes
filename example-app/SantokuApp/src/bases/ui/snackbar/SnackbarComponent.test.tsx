import {act, render, screen} from '@testing-library/react-native';
import React from 'react';
import {TextStyle, ViewStyle} from 'react-native';
import {ReactTestInstance} from 'react-test-renderer';

import {SnackbarComponent} from './SnackbarComponent';

// If advancing a timer changes the state of a component, the timer must be run within an act.
// However, since act is `Thenable`, ESLint will issue a warning if you do not do something like await.
// For convenience, disable the relevant rule in this file.
/* eslint-disable @typescript-eslint/no-floating-promises */

jest.useFakeTimers('modern');

function getStyle<T>(instance: ReactTestInstance) {
  return instance.props.style as T;
}

const AUTO_HIDE_DURATION = 4000;
const FADE_IN_DURATION = 1000;
const FADE_OUT_DURATION = 1000;
const FORCE_FADE_OUT_DURATION = 300;
const HIDE_FADE_OUT_DURATION = 300;

describe('SnackbarComponent', () => {
  it('SnackbarComponentが正常にrenderできることを確認', () => {
    render(<SnackbarComponent message="テストメッセージ" />);

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();

    expect(getStyle<ViewStyle>(screen.getByTestId('snackbarAnimatedView')).opacity).toBe(0);
    expect(screen).toMatchSnapshot('render直後');

    act(() => {
      jest.advanceTimersByTime(FADE_IN_DURATION);
    });
    expect(getStyle<ViewStyle>(screen.getByTestId('snackbarAnimatedView')).opacity).toBe(1);
    expect(screen).toMatchSnapshot('フェードイン後');

    act(() => {
      jest.advanceTimersByTime(AUTO_HIDE_DURATION + FADE_OUT_DURATION);
    });
    expect(screen.queryByTestId('snackbarAnimatedView')).toBeNull();
    expect(screen).toMatchSnapshot('フェードアウト後');
  });

  it('SnackbarComponent表示中にpropsが更新された場合、フェードアウト後に更新後のpropsでSnackbarComponentが表示されることを確認', () => {
    render(<SnackbarComponent message="初回" />);

    expect(screen.queryByText('初回')).not.toBeNull();

    screen.update(<SnackbarComponent message="２回目" />);

    expect(screen.queryByText('初回')).not.toBeNull();

    act(() => {
      jest.advanceTimersByTime(FORCE_FADE_OUT_DURATION);
    });

    expect(screen.queryByText('初回')).toBeNull();
    expect(screen.queryByText('２回目')).not.toBeNull();
  });

  it('SnackbarComponentの表示中に２連続propsが更新された場合、後で更新した方のpropsでSnackbarComponentが表示されることを確認', () => {
    render(<SnackbarComponent message="初回" />);

    expect(screen.queryByText('初回')).not.toBeNull();

    screen.update(<SnackbarComponent message="２回目" />);

    screen.update(<SnackbarComponent message="３回目" />);

    act(() => {
      jest.advanceTimersByTime(FORCE_FADE_OUT_DURATION);
    });

    expect(screen.queryByText('３回目')).not.toBeNull();
  });

  it('SnackbarComponentの表示後に同一のpropsを指定した場合、SnackbarComponentが表示されないことを確認', () => {
    const props = {
      message: 'テストメッセージ',
      messageTextStyle: {color: 'white'},
      style: {backgroundColor: 'aqua'},
      positionStyle: {minHeight: 60},
      actionText: 'close',
      actionHandler: jest.fn(),
      actionTextStyle: {color: 'red'},
      autoHideDuration: 100,
      fadeInDuration: 200,
      fadeOutDuration: 300,
      forceFadeOutDuration: 400,
      timestamp: Date.now(),
    };
    render(<SnackbarComponent {...props} />);

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();

    act(() => {
      jest.advanceTimersByTime(FADE_IN_DURATION + AUTO_HIDE_DURATION + FADE_OUT_DURATION);
    });

    screen.update(<SnackbarComponent {...props} />);

    expect(screen.queryByText('テストメッセージ')).toBeNull();
  });

  it('SnackbarComponentの表示後にTimestamp以外同一のpropsを指定した場合、SnackbarComponentが表示されることを確認', () => {
    const props = {
      message: 'テストメッセージ',
      messageTextStyle: {color: 'white'},
      style: {backgroundColor: 'aqua'},
      positionStyle: {minHeight: 60},
      actionText: 'close',
      actionHandler: jest.fn(),
      actionTextStyle: {color: 'red'},
      autoHideDuration: 100,
      fadeInDuration: 200,
      fadeOutDuration: 300,
      forceFadeOutDuration: 400,
    };
    render(<SnackbarComponent {...props} timestamp={Date.now()} />);

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();

    act(() => {
      jest.advanceTimersByTime(FADE_IN_DURATION + AUTO_HIDE_DURATION + FADE_OUT_DURATION);
    });

    screen.update(<SnackbarComponent {...props} timestamp={Date.now()} />);

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();
  });

  it('SnackbarComponent表示中にpropsでhideを指定した場合、SnackbarComponentが消えることを確認', () => {
    render(<SnackbarComponent message="テストメッセージ" />);

    jest.advanceTimersByTime(FADE_IN_DURATION);
    expect(screen.queryByText('テストメッセージ')).not.toBeNull();

    screen.update(<SnackbarComponent message="テストメッセージ" hide />);

    act(() => {
      jest.advanceTimersByTime(HIDE_FADE_OUT_DURATION);
    });

    expect(screen.queryByText('テストメッセージ')).toBeNull();
  });

  it('SnackbarComponentに指定したpropsがrenderに反映されていることを確認', () => {
    /**
     * 下記項目は対象外
     * - autoHideDuration / fadeInDuration / fadeOutDuration / forceFadeOutDuration / hideFadeOutDuration
     *     テストコードでアニメーション時間を調整できない
     * - actionHandler
     *     setVisibleSnackbarProps()でpropsをセットした時点で、指定したモック関数ではなくなっており、追跡できない
     * - hide
     *     テスト済
     */
    render(
      <SnackbarComponent
        message="テストメッセージ"
        messageTextStyle={{color: 'black'}}
        style={{backgroundColor: 'red'}}
        positionStyle={{bottom: 50}}
        actionText="閉じる"
        actionHandler={() => {}}
        actionTextStyle={{color: 'blue'}}
      />,
    );

    jest.advanceTimersByTime(FADE_IN_DURATION);

    expect(screen.queryByText('テストメッセージ')).not.toBeNull();
    expect(getStyle<TextStyle>(screen.getByText('テストメッセージ')).color).toBe('black');
    expect(getStyle<ViewStyle>(screen.getByTestId('snackbarStyleView')).backgroundColor).toBe('red');
    expect(getStyle<ViewStyle>(screen.getByTestId('snackbarAnimatedView')).bottom).toBe(50);
    expect(screen.queryByText('閉じる')).not.toBeNull();
    expect(getStyle<TextStyle>(screen.getByText('閉じる')).color).toBe('blue');
  });
});
