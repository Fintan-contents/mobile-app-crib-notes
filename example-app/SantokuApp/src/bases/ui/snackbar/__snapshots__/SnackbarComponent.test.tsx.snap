// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackbarComponent SnackbarComponentが正常にrenderできることを確認: render直後 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "bottom": 20,
      "flex": 1,
      "minHeight": 48,
      "opacity": 0,
      "paddingHorizontal": 10,
      "position": "absolute",
      "width": "100%",
    }
  }
  testID="snackbarAnimatedView"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#393939",
        "borderRadius": 5,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
    testID="snackbarStyleView"
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingLeft": 20,
          "paddingRight": 10,
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        テストメッセージ
      </Text>
    </View>
  </View>
</View>
`;

exports[`SnackbarComponent SnackbarComponentが正常にrenderできることを確認: フェードアウト後 1`] = `null`;

exports[`SnackbarComponent SnackbarComponentが正常にrenderできることを確認: フェードイン後 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "bottom": 20,
      "flex": 1,
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 10,
      "position": "absolute",
      "width": "100%",
    }
  }
  testID="snackbarAnimatedView"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#393939",
        "borderRadius": 5,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
    testID="snackbarStyleView"
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingLeft": 20,
          "paddingRight": 10,
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        テストメッセージ
      </Text>
    </View>
  </View>
</View>
`;
